src/Backend.idr

```diff
@@ -76,6 +76,14 @@ interface Backend lang where
 generate : (lang: Type) -> Backend lang => {n: Nat} -> TDef n -> Doc
 generate lang {n} td = vsep2 . map (generateCode) . generateTyDefs {lang} (freshEnv {lang} n) $ td
 
+interface NewBackend def type | def where
```

Comment:
Maybe its nice to add a short doc comment here that describes what `msgType`, `typedefs` and `source` are supposed to do?